milli/src/search/criteria/geo.rs

```diff
@@ -104,12 +104,12 @@ impl Criterion for Geo<'_> {
 
 fn geo_point(
     rtree: &RTree<GeoPoint>,
-    candidates: RoaringBitmap,
+    mut candidates: RoaringBitmap,
     point: [f64; 2],
 ) -> Box<dyn Iterator<Item = RoaringBitmap>> {
     let results = rtree
         .nearest_neighbor_iter(&point)
-        .filter_map(move |point| candidates.contains(point.data).then(|| point.data))
+        .filter_map(move |point| candidates.remove(point.data).then(|| point.data))
```

Comment:
This doesn't make the iteration stop, you should add something like a `take_while` or even better: rewrite this iterator combination as a for loop, instead.

Reply:
oops yes, is this ok?
https://github.com/meilisearch/milli/pull/322/commits/91ce4d1721765cb5862f6cf531fea0d2021021f0